src/platform/utilities/tests/ui/index.unit.spec.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { expect } from 'chai';
+
+import { isReactComponent } from '../../ui';
+
+describe('isReactComponent', () => {
+  it('should return true for function component', () => {
+    expect(isReactComponent(() => <div />)).to.be.true;
+  });
+  it('should return true for class component', () => {
+    class MyComponent extends React.Component {
+      constructor() {
+        super();
+        this.state = { test: true };
+      }
+      render() {
+        return <div />;
+      }
+    }
+    expect(isReactComponent(MyComponent)).to.be.true;
+  });
+  it('should return true for memoized component', () => {
+    expect(isReactComponent(React.memo(() => <div />))).to.be.true;
+  });
+  it('should return false for regular element', () => {
+    expect(isReactComponent(<div />)).to.be.false;
+  });
+  it('should return false for fragment', () => {
+    expect(isReactComponent(<>Test</>)).to.be.false;
+  });
+  it('should return false for string', () => {
+    expect(isReactComponent('Test')).to.be.false;
+  });
+  it('should return false for number', () => {
```

Comment:
Would it be useful to have a test for null or undefined here?

Reply:
Couldn't hurt!